pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2110,6 +2165,50 @@ static __always_inline unsigned short get_inode_mode_from_fd(u64 fd)
     return READ_KERN(f_inode->i_mode);
 }
 
+static __always_inline void detect_hooked_syscalls(event_data_t *data){
+    int key = 1;
+    u64 *table_ptr = bpf_map_lookup_elem(&hooked_syscalls_map, (void *)&key);
+    if (table_ptr == NULL){
+        return ;
+    }
+
+    unsigned long *syscall_table_addr = (unsigned long*) *table_ptr;
+    u64 idx;
+    u64* snum_ptr;                  // pointer to syscall_number
+    u64 snum;                       // syscall_number
+    unsigned long saddr = 0;        // syscall address
+    int number_of_hooked_syscalls = 0;
+    #if defined(bpf_target_x86)
+        number_of_hooked_syscalls = NUMBER_OF_SYSCALLS_X86;
+        u64 syscall_address[NUMBER_OF_SYSCALLS_X86];
+    #elif defined(bpf_target_arm64)
+        number_of_hooked_syscalls = NUMBER_OF_SYSCALLS_ARM;
+        u64 syscall_address[NUMBER_OF_SYSCALLS_ARM];
+    #else
+
+        return
+    #endif
+
+    __builtin_memset(syscall_address, 0, sizeof(syscall_address));
+    #pragma unroll
+    for(int i =0; i < number_of_hooked_syscalls; i++){
+        idx = i+2;
+        snum_ptr = bpf_map_lookup_elem(&hooked_syscalls_map, (void *)&idx);
+        if (snum_ptr == NULL){
+            return;
```

Comment:
Why do you return here?
```suggestion
            continue;
```